### PR TITLE
Convert `ErrorBar` e2e tests to use `MockCallAdapterState`

### PR DESCRIPTION
**Pull request by prprabhu-ms**
# What

Replaces a test specific `TestCallingState` data structure with `MockCallAdapterState`. The new type closely follows `CallAdapterState`. The only difference is that video element views `HTMLElement` is left unset, to be populated by the test app.

# Why

Makes it easier to write more e2e test cases by avoiding having to extend `TestCallingState` repeatedly.

### CODE REVIEW

File: packages/react-composites/tests/browser/call/MockCallAdapterState.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+} from '@internal/calling-stateful-client';
+import type { CallAdapterState } from '../../../src';
+
+/**
+ * A slight modification of {@link CallAdapterState} for initializing the
+ * {@link MockCallAdapter} for hermetic e2e tests.
+ *
+ * - HTMLElement views are replaced with placeholders to be populated
+ *   by the test app.
+ */
```
Comment (mgamis-msft) [inline packages/react-composites/tests/browser/call/MockCallAdapterState.ts:19]: Thanks for creating these Mock interfaces. Now it is tightly coupled with the state interfaces of in 'calling-stateful-client'

File: packages/react-composites/tests/browser/call/app/mocks/MockCallAdapter.ts [hunk 2]
```diff
@@ -2,56 +2,33 @@
...
-          media: { latest: { ...initialState.call.diagnostics.media.latest, ...testState.diagnostics?.media } }
-        };
-      }
-    }
-    if (testState.page) {
-      initialState.page = testState.page;
+  /**
+   * `testState` is deprecated. Use `initialState` instead.
+   */
+  constructor(testState?: TestCallingState, initialState?: MockCallAdapterState) {
```
Comment (mgamis-msft) [inline packages/react-composites/tests/browser/call/app/mocks/MockCallAdapter.ts:23]: Thanks for adding the state as a constuctor argument. Looks much cleaner now :)

**Review by mgamis-msft — APPROVED** [review]

**Review by carocao-msft — APPROVED** [review]

File: packages/react-composites/tests/browser/call/app/HermeticApp.tsx [hunk 3]
```diff
@@ -6,30 +6,23 @@ import React, { useState, useEffect } from 'react';
 import { CallAdapter } from '../../../../src';
 import { MockCallAdapter } from './mocks/MockCallAdapter';
 import { QueryArgs } from './QueryArgs';
-import { TestCallingState } from '../TestCallingState';
 import { BaseApp } from './BaseApp';
 
-/** @internal */
-export function HermeticApp(props: { queryArgs: HermeticAppQueryArgs }): JSX.Element {
+export function HermeticApp(props: { queryArgs: QueryArgs }): JSX.Element {
```
Comment (dmceachernmsft) [inline packages/react-composites/tests/browser/call/app/HermeticApp.tsx:11]: Maybe a silly question, but why is it called a `HermeticApp`?

File: packages/react-composites/tests/browser/call/app/mocks/MockCallAdapter.ts [hunk 4]
```diff
@@ -2,56 +2,33 @@
...
-import { CallAdapter, CallAdapterState } from '../../../../../src';
-import { TestCallingState, TestRemoteParticipant } from '../../TestCallingState';
+import type { CallAdapter, CallAdapterState } from '../../../../../src';
+import type { MockCallAdapterState } from '../../MockCallAdapterState';
+import type { TestCallingState, TestRemoteParticipant } from '../../TestCallingState';
 
 /**
  * Mock class that implements CallAdapter interface for UI snapshot tests. The handler implementation is currently limited so
  * some composite behaviour will not work like clicking the 'Start Call' button in the Configuration page. The usage of
  * MockCallAdapter is intended to talk snapshot based only on the state of a CallAdapter.
```
Comment (dmceachernmsft) [inline packages/react-composites/tests/browser/call/app/mocks/MockCallAdapter.ts:17]: nit: take

File: packages/react-composites/tests/browser/call/hermetic/ErrorBar.test.ts [hunk 5]
```diff
@@ -1,53 +1,50 @@
...
-            target: 'Call.stopVideo',
-            innerError: new Error('Inner error of failure to stop video')
-          }
-        }
-      })
-    );
+    const initialState = defaultMockCallAdapterState();
+    initialState.latestErrors = {
+      'Call.unmute': {
+        timestamp: new Date(),
```
Comment (dmceachernmsft) [inline packages/react-composites/tests/browser/call/hermetic/ErrorBar.test.ts:33]: Do these time stamps need to be real times? or would just giving them a dummy value be better?

**Review by dmceachernmsft — APPROVED** [review]

File: packages/react-composites/tests/browser/call/hermetic/ErrorBar.test.ts (hunk 5 above)
Comment (prprabhu-ms) [reply packages/react-composites/tests/browser/call/hermetic/ErrorBar.test.ts:33]: I guess not necessary, but I'm avoiding any semantic changes in this PR.

**Pull request merged**